dora/core/server/worker/src/main/java/alluxio/worker/dora/PagedDoraWorker.java

```diff
@@ -306,11 +301,14 @@ private void registerToMaster() throws IOException {
     RetryPolicy retry = RetryUtils.defaultWorkerMasterClientRetry();
     while (true) {
       try (PooledResource<BlockMasterClient> bmc = mBlockMasterClientPool.acquireCloseable()) {
-        mWorkerId.set(bmc.get().getId(mAddress));
+        long id = bmc.get().getId(mAddress);
+        LOG.debug("Obtained worker id {} from master", id);
+        WorkerIdentity identity = WorkerIdentity.ParserV0.INSTANCE.fromLong(id);
+        mWorkerId.set(identity);
```

Comment:
So it seems that master is still required though user can use ETCD as worker discovery service, righit?

Reply:
right

Reply:
Sorry, maybe I missed something.
Do you mean, `master` is still required to run in a Alluxio cluster even when we have ETCD service?

Reply:
For worker discovery purposes, either etcd or master is required, but not both.
For other features like PDDM, master is required.
If the user needs PDDM, then yes, etcd and master are required at the same time.